### PR TITLE
[Tabs] Font/Color Scheme support for tab bar.

### LINKED ISSUE #3086: [Tabs] Font Scheme
As an app developer, I want to theme the typography of all my Tabs to reflect my brand's typography.

### LINKED ISSUE #3087: [Tabs] Color Scheme
As an app developer, I want to theme the colors of all my Tabs to reflect my brand colors.

### LINKED ISSUE #3089: [Tabs] MDCTabBarColorThemer does not check if a color scheme implements optional properties
### Overview
MDCTabBarColorThemer uses `primaryLightColor` and [`primaryDarkColor`](https://github.com/material-components/material-components-ios/blob/develop/components/Tabs/src/ColorThemer/MDCTabBarColorThemer.m#L23) without checking if the objects responds to the selectors. This can cause crashes if an [`<MDCColorScheme>`](https://github.com/material-components/material-components-ios/blob/develop/components/Themes/src/MDCColorScheme.h#L28) object is passed that does not implement these optional properties.

### Reproduction steps

```objectivec
@interface FakeScheme : NSObject <MDCColorScheme>
@end

@implementation FakeScheme
- (UIColor *)primaryColor {
  return UIColor.whiteColor;
}
@end
@interface MDCTabBarTests : XCTestCase
@end

@implementation MDCTabBarTests
- (void)testScheme {
  FakeScheme *scheme = [[FakeScheme alloc] init];
  MDCTabBar *bar = [[MDCTabBar alloc] init];
  [MDCTabBarColorThemer applyColorScheme:scheme toTabBar:bar];
}
@end
```

### Version number
46.1.1

### PR DESCRIPTION
**Pull request by mohammadcazig**
Added Color and Font Themer for TabBar. 
Close #3086 
Close #3087 
Close #3089

### CODE REVIEW

File: components/Tabs/src/FontThemer/MDCTabBarFontThemer.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
```
Comment (romoore) [inline components/Tabs/src/FontThemer/MDCTabBarFontThemer.h]: Copyright date: s/2017/2018/

File: components/Tabs/src/FontThemer/MDCTabBarFontThemer.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ Applies a font scheme to theme a MDCTabBar. Use a UIAppearance proxy to apply a font scheme to
+ all instances of MDCTabBar.
+
+ @param fontScheme The font scheme to apply to MDCTabBar.
+ @param tabBar A MDCTabBar instance to apply a color scheme.
+ */
++ (void)applyFontScheme:(nonnull id<MDCFontScheme>)fontScheme
+               toTabBar:(nonnull MDCTabBar *)tabBar;
```
Comment (romoore) [inline components/Tabs/src/FontThemer/MDCTabBarFontThemer.h]: Nit: I think the `tabBar` property can safely be `nil`.

File: components/Tabs/src/FontThemer/MDCTabBarFontThemer.m [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
```
Comment (romoore) [inline components/Tabs/src/FontThemer/MDCTabBarFontThemer.m]: Copyright: s/2017/2018/

File: components/Tabs/src/FontThemer/MDCTabBarFontThemer.m [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCTabBarFontThemer.h"
+
+@implementation MDCTabBarFontThemer
+
++ (void)applyFontScheme:(nonnull id<MDCFontScheme>)fontScheme
```
Comment (romoore) [inline components/Tabs/src/FontThemer/MDCTabBarFontThemer.m]: Nit: Consider removing the nullability annotations from the implementation.

File: components/Tabs/tests/unit/MDCTabBarColorSchemeTests.m [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  return [UIColor darkGrayColor];
+}
+
+- (UIColor *)primaryLightColor {
+  return [UIColor yellowColor];
+}
+
+@end
+
+@interface MDCTabBarColorSchemeTests : XCTestCase
```
Comment (romoore) [inline components/Tabs/tests/unit/MDCTabBarColorSchemeTests.m]: Naming: This looks like an MDCTabBarColorThemer test, not a scheme test. Consider renaming to `MDCTabBarColorThemerTests`

File: components/Tabs/tests/unit/MDCTabBarFontSchemeTests.m [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+ limitations under the License.
+ */
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialTabs.h"
+#import "MaterialThemes.h"
+#import "MDCTabBarFontThemer.h"
+
+@interface MDCTabBarFontSchemeTests : XCTestCase
```
Comment (romoore) [inline components/Tabs/tests/unit/MDCTabBarFontSchemeTests.m]: Same: This looks like a themer test and not a scheme test.

**Review by romoore — APPROVED** [review]

**Comment by romoore** [comment]
Looks like this needs a rebase?

File: components/Tabs/BUILD [hunk 7]
```diff
@@ -63,6 +63,22 @@ mdc_objc_library(
...
+        "UIKit",
+    ],
+    deps = [
+        ":Tabs",
+        "//components/Themes",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+
```
Comment (romoore) [inline components/Tabs/BUILD]: Nit: Remove extra empty line.

File: components/Tabs/src/FontThemer/MDCTabBarFontThemer.h [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#import "MaterialThemes.h"
+#import "MaterialTabs.h"
```
Comment (romoore) [inline components/Tabs/src/FontThemer/MDCTabBarFontThemer.h]: Nit: Order of imports.

File: components/Tabs/tests/unit/MDCTabBarColorThemerTests.m [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+@end
+
+@interface MDCTabBarColorThemerTests : XCTestCase
+
+@end
+
+@implementation MDCTabBarColorThemerTests
+
+- (void)testTabBarColorThemerApplyColorSchemeProperly {
```
Comment (romoore) [inline components/Tabs/tests/unit/MDCTabBarColorThemerTests.m:61]: Please add a test (or a TODO [referencing the backlog issue](https://github.com/material-components/material-components-ios/issues/3089)) to test a color scheme that has only the `primaryColor` property and how the themer will behave.

**Review by romoore — APPROVED** [review]

**Comment by romoore** [comment]
LGTM!

**Review by ianegordon — APPROVED** [review]
Please merge once Travis is happy.

**Pull request merged**